core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -191,6 +211,151 @@ public void process(WatchedEvent event) {
         externalWatcher.process(event);
       }
     }
+
+    private synchronized void processTableConfigurationItem(WatchedEvent event) {
+
+      if (event.getPath() == null || event.getPath().isEmpty())
+        return;
+
+      Matcher tableConfigsMatcher = TABLE_CONFIGS_PATTERN.matcher(event.getPath());
+      if (!tableConfigsMatcher.matches()) {
+        return;
+      }
+
+      try {
+        Stat versionStat = getZooKeeper().exists(event.getPath(), watcher);
+        if (versionStat == null) {
+          return;
+        }
+
+        byte[] configToRefresh = getZooKeeper().getData(event.getPath(), true, versionStat);
+
+        if (configToRefresh == null) {
+          return;
+        }
+
+        if ((versionStat != null) && (versionStat.getVersion() - lastTableConfigVersion == 1)) {
+          lastTableConfigVersion = versionStat.getVersion();
+          refreshTableConfig(configToRefresh);
+          if (log.isTraceEnabled()) {
+            log.trace("Successfully refreshed table config (the most efficient way way): "
```

Comment:
The most efficient way way ?

Reply:
I will change that to something like " by only refreshing the updated table configuration."